src/models/lineChart.js

```diff
@@ -306,6 +351,30 @@ nv.models.lineChart = function() {
                 chart.update();
             });
 
+            lines.dispatch.on('elementMouseover.tooltip', function(evt) {
+                tooltip.data(evt).position(evt.pos).hidden(false);
+
+                if(showDistX){
+                    container.select('.nv-distributionX .nv-series-' + evt.seriesIndex + ' .nv-distx-' + evt.pointIndex)
+                        .attr('y1', evt.relativePos[1] - availableHeight);
+                }
+                if(showDistY){
+                    container.select('.nv-distributionY .nv-series-' + evt.seriesIndex + ' .nv-disty-' + evt.pointIndex)
+                        .attr('x2', evt.relativePos[0] + distX.size());
+                }
+            });
+
+            lines.dispatch.on('elementMouseout.tooltip', function(evt) {
+                tooltip.hidden(true);
+                if(showDistX){
+                    container.select('.nv-distributionX .nv-series-' + evt.seriesIndex + ' .nv-distx-' + evt.pointIndex)
+                        .attr('y1', 0);
+                }
+                if(showDistY){
+                    container.select('.nv-distributionY .nv-series-' + evt.seriesIndex + ' .nv-disty-' + evt.pointIndex)
+                        .attr('x2', distY.size());
+                }
+            });
```

Comment:
I just make it like the scatterChart. You are right , I will move it
outside the chart scope. Also I will move the one in scatterChart outside
the scope too. And I will do some optimization on the scatterChart.

2015-07-27 19:18 GMT+02:00 liquidpele <notifications@github.com>:

> In src/models/lineChart.js
> <https://github.com/novus/nvd3/pull/1170#discussion_r35561321>:
>
> > +                    container.select('.nv-distributionY .nv-series-' + evt.seriesIndex + ' .nv-disty-' + evt.pointIndex)
> > +                        .attr('x2', evt.relativePos[0] + distX.size());
> > +                }
> > +            });
> > +
> > +            lines.dispatch.on('elementMouseout.tooltip', function(evt) {
> > +                tooltip.hidden(true);
> > +                if(showDistX){
> > +                    container.select('.nv-distributionX .nv-series-' + evt.seriesIndex + ' .nv-distx-' + evt.pointIndex)
> > +                        .attr('y1', 0);
> > +                }
> > +                if(showDistY){
> > +                    container.select('.nv-distributionY .nv-series-' + evt.seriesIndex + ' .nv-disty-' + evt.pointIndex)
> > +                        .attr('x2', distY.size());
> > +                }
> > +            });
>
> Why are you moving these functions here? They'll get overwritten every
> chart.update() in this case.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novus/nvd3/pull/1170/files#r35561321>.
>